pkg/osquery/enrollment_details.go

```diff
@@ -31,22 +37,23 @@ func getEnrollDetails(ctx context.Context, osquerydPath string) (service.Enrollm
 		return details, errors.New("Skipping enrollment details")
 	}
 
+	details.OSPlatform = runtime.GOOS
+	details.OSPlatformLike = runtime.GOOS
+
 	// If the binary doesn't exist, bail out early.
 	if info, err := os.Stat(osquerydPath); os.IsNotExist(err) {
-		return details, fmt.Errorf("no binary at %s", osquerydPath)
+		return details, fmt.Errorf("%w: no binary at %s:", osqueryNotFoundErr{}, osquerydPath)
```

Comment:
I don't think we should return content and errors. It feels like it breaks the usual go idioms. 

My inclination is to log this, and then use the best we have. 

Slightly longer term, I'd like to calculate all the enroll details when either launcher or osquery restart, and then store them in knapsack for subsequent usage. If there's some issue with osquery, I'd want to allow things to get the go only ones, and then fill in the osquery ones when possible

Reply:
Could solve this by doing e.g. `return details, fmt.Errorf("%w: no binary", osqueryNotFoundErr{})`, and then in extension.go log `osqPath` at the same time that we log this error.

Reply:
> I don't think we should return content and errors. It feels like it breaks the usual go idioms.

I see what you mean, this kind of implies that `getEnrollDetails` will never return an error because we always have access to the runtime values?